docs/development/design/organizations.rst

```diff
@@ -0,0 +1,148 @@
+#. Integrate the code from the community site to the new code.
+
+We should start by removing unused features and dead code from the organizations in the corporate site,
+and simplify existing code if possible (some of this was already done).
+Another simplification that can be done is:
+make the relationship between the models ``Organization`` and ``Project`` one to many
+(currently many to many).
+
+Isolate/separate the models to be moved from the ones that aren't going to be moved.
+We should move the models that aren't going to me moved to another app.
+
+- Plan
+- PlanFeature
+- Subscription
+
+This app can be named *subscriptions*.
+We can get around the table names and migrations by setting the explicitly the table name to ``organizations_<model>``,
+and doing a fake migration.
+Following suggestions in https://stackoverflow.com/questions/48860227/moving-multiple-models-from-one-django-app-to-another.
+Code related to subscriptions should be moved out from the organizations app.
+
+After that, it should be easier to move the organizations *app* (or part of it)
+to the community site (and no changes to table names would be required).
+
+We start by moving the models.
+
+- Organization
+- OrganizationOwner
+- Team
+- TeamInvite
+- TeamMember
+
+Migrations aren't moved, since all current migrations depend on other models that aren't
+going to be moved.
+In the community site we run an initial migration,
+for the corporate site we run a fake migration.
+
+For managers and querysets that depend on subscriptions,
+we can use our pattern to make overridable classes (inheriting from ``SettingsOverrideObject``).
+
+Templates, urls, views, forms, notifications, signals, tasks can be moved later
+(we just need to make use of the models from the ``readthedocs.organizations`` module).
+
+If we decide to integrate organizations in the community site,
+we can enable the app.
+
+After the app is moved,
+we can move more code that depends on organizations to the community site.
+
+Namespace
+---------
+
+Currently we use the project's slug as namespace,
+in the commercial site we use the combination of ``organization.slug`` + ``project.slug`` as namespace.
+
+For the community site probably this approach isn't the best,
+since we always serve docs publicly from ``slug.readthedocs.io``.
+And most of the users don't have a custom domain.
+
+We could keep the current behavior for the community site and use ``organization.slug`` + ``project.slug`` for the corporate site,
+since in the corporate site we don't care so much about a unique namespace between all users, but a unique namespace per organization.
+We can refactor the way we get the namespace to be more easy to manage in both sites.
+
+Future Changes
+--------------
+
+Changes that aren't needed immediately after the migration,
+but that should be done:
+
+Edit the current querysets, modify/add UI elements, and add new endpoints to the API (v3 only).
```

Comment:
Modifying the UI is not going to be a future change, it will be tied directly to this work. We should probably have an idea of at least what UI components are likely to require changes. We won't be able to just copy the UI bits into community as we've discussed structural changes -- supporting user projects and organization projects for instance.

Reply:
I just put some pieces not sure if we want to copy most of the UI from .com or make something different (especially for listing and detail of orgs)?

Reply:
If we're not going to enable this on the community site yet, and we're going to do a two step process -- move the code over, and then sometime later actually enable organizations on the community codebase -- then we can worry about UI later.

Do we want to source templates for organizations from the commercial repo or move to the community repo?

Reply:
Yes, it's a two step migration.

> Do we want to source templates for organizations from the commercial repo or move to the community repo?

Good question, I think we can move the urls/views later, that way the templates can live in .com in the meantime and is easy to _not enable_ organization in the community site.